src/Network/NWFramer.cs

```diff
@@ -238,23 +250,25 @@ static void TrampolineCleanupHandler (IntPtr block, OS_nw_framer framer)
 		static extern void nw_framer_schedule_wakeup (OS_nw_framer framer, ulong milliseconds);
 
 		public void ScheduleWakeup (ulong milliseconds) => nw_framer_schedule_wakeup (GetCheckedHandle (), milliseconds);
+		*/
 
 		[DllImport (Constants.NetworkLibrary)]
 		static extern OS_nw_protocol_metadata nw_framer_protocol_create_message (OS_nw_protocol_definition definition);
 
-		public static NWFramer CreateMessage (NWProtocolDefinition protocolDefinition)
+		public static NWFramerMessage CreateMessage (NWProtocolDefinition protocolDefinition)
 		{
 			if (protocolDefinition == null)
 				throw new ArgumentNullException (nameof (protocolDefinition));
-			return new NWFramer (nw_framer_protocol_create_message (protocolDefinition.Handle), owns: true);
+			return new NWFramerMessage (nw_framer_protocol_create_message (protocolDefinition.Handle), owns: true);
 		}
 
 		[DllImport (Constants.NetworkLibrary)]
 		static extern OS_nw_protocol_metadata nw_framer_message_create (OS_nw_framer framer);
 
-		public NWProtocolMetadata CreateMessage ()
-			=> new NWFramer (nw_framer_message_create (GetCheckedHandle ()), owns: true);
+		public NWFramerMessage CreateMessage ()
```

Comment:
This is not a static method, but a instance one. Needs the pointer to the framer. The one above is kept in the same class since it makes sense to group them. Would look funny to create a message in two different classes.

Reply:
ok, I think it's better not have `CreateMessage` as both static and instance methods (so this one is fine)

Reply:
Sorry, I think I changed my mind. You're right it's not `static` but that's how you chose to expose it, not how Apple did it.

The function is named `nw_framer_message_create` so the prefix makes me think it should be inside `NWFramerMessage` not here in `NWFramer`

IOW it's `nw_framer_message_create` -> `NWFramerMessage.Create`
and not `NWFramer.MessageCreate`